### ISSUE TITLE
Support Mono/LLVM as a runtime/jit

### ISSUE THREAD

**Issue by damageboy** [PerfDotNet/BenchmarkDotNet#226]
Mono has a special "mode" called [Mono/LLVM](http://www.mono-project.com/docs/advanced/mono-llvm/).

The basic premise with Mono/LLVM is that mono JIT is swapped out for LLVM as a JIT.

It would be great to add support for it in BDN to see how LLVM affects performance (if at all :))

I don't know if there is a programmatic way to detect LLVM support, but here is a hack way...

On my machine, I have both versions of mono installed (With+Without LLVM), here is what `mono -V` reports:

Without LLVM:
```
Mono JIT compiler version 4.4.1 (Nightly 4.4.1.0/4747417 Wed Jun 22 11:14:49 UTC 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen

```

With LLVM:
```
Mono JIT compiler version 4.4.1 (explicit/4747417 Wed Jun 22 19:25:42 IDT 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          yes(3.6.0svn-mono-master/9f79399)
	GC:            sgen
```

**Comment by adamsitnik** [comment]
I will be very happy to implement it.

@AndreyAkinshin should it be a new value for our Jit enum or a whole new Toolchain?

**Comment by AndreyAkinshin** [comment]
@adamsitnik, oh, our Jobs become very complex.
I suggest to create a string property which will pass to mono (or dotnet/cli) as a list of arguments. This flexible approach covers a lot of different scenarios.

**Comment by adamsitnik** [comment]
@damageboy Ok, I think I have it. The problem I have is that I can't test it on Windows ;/

Could you try to test it? You need to download 0.9.8.56 from our dev feed:
`<add key="appveyor-bdn" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />`
and compare some benchmarks with Job.Mono and Job.MonoLLVM.

**Comment by AndreyAkinshin** [comment]
@adamsitnik, can we rename `LLVM` to `Llvm`?

**Comment by adamsitnik** [comment]
@AndreyAkinshin sure! I have found some [nice explanation](http://stackoverflow.com/a/15526526/5852046) when to use camelcase for Acronyms.

I don't like to create very small commits with single rename so I will rename it with some other commit soon. And don't hesitate to refactor my code or comment it, all code reviews are welcome!

**Comment by AndreyAkinshin** [comment]
@adamsitnik, what the current state of this issue?

**Comment by adamsitnik** [comment]
@AndreyAkinshin 

TL;DR Implemented, but not tested in real life scenario.

I have implemented this based on what I have found in documentation. It's quite simple feature, a matter of adding extra parameter to mono. The problem is that I don't have access to any Unix machine now so I am not able to test it.

**Comment by AndreyAkinshin** [comment]
Ok, I will check it.

**Comment by AndreyAkinshin** [comment]
I failed to build mono+llvm, there are some troubles on my linux machine. However, I looked at the source code, it looks fine. =) Let's assumed that llvm works until somebody sends a bug report. =)
